app/access/models.py

```diff
@@ -0,0 +1,16 @@
+from django.db import models
+from django.utils.translation import pgettext_lazy as _
+
+
+class User(models.Model):
+
+    _context = "User model"
+
+    def __str__(self) -> str:
+        return str(self.first_name) + str(self.last_name)
+
+    first_name = models.CharField(_(_context, "First name"))
+    last_name = models.CharField(_(_context, "Last name"))
+    email = models.EmailField(_(_context, "Email"))
```

Comment:
Seeing from the cognito implementation of @msom I think you need `username` here too. Afaik the username will also be the unique identifier in the entire system? Though I'm not quite sure anymore if or what we agreed here on.

Reply:
I would also use a designated field, independent of the email. Nonetheless, I think email needs to be unique.

Reply:
Hm ok, I was not aware of that there is already [a PR with for the interface with cognito](https://github.com/geoadmin/service-control/pull/29).

Having a dedicated `username` sounds like a good idea for when emails change. I can add that.

I can also try to add a constraint that the email should be unique in the DB. From what I understand, setting [`unique=True`](https://docs.djangoproject.com/en/5.1/ref/models/fields/#unique) in the constructor should be enough.

Reply:
The cognito PR is a draft and I will adapt it to what you'll implement.

Reply:
The username should be unique as well.

Reply:
I agree we should not use e-mail as username because we probably want to be able to change e-mail. I have no strong opinion on e-mail uniqueness. The use case @boecklic describes can easily work around that with [sub-addressing](https://en.wikipedia.org/wiki/Email_address#Sub-addressing) on their side (i.e. foo+bar@baz.quux) but I guess it's nice to let users use whatever address they control, be it duplicated or not.

However I am unclear why we need first_name and last_name. What about a [single full_name field](https://www.kalzumeus.com/2010/06/17/falsehoods-programmers-believe-about-names/)? However, given enough diverse users, deriving a username from the full name might be more tricky than it seems. Some entropy and checking against existing users will be required in any case. Would be nice to let the user pick their username even if we suggest one by default.

Reply:
I find it useful to be able to clearly distinguish between first_ and last_name, e.g. for ordering and be sure we have both. If it's just a fullname field we'll have a mix of 'first_name last_name', 'first_name', 'last_name first_name' etc.

Reply:
Thanks for your inputs.

For the moment, I change the following:
- `username`: Primary key, no defaults for simplicity. We can add that later.
- `email`: Unchanged, no requirement for uniqueness.
- `first_name`, `last_name`: Leave it as it is. I like the single `full_name` though.

I propose we quickly discuss the details in person tomorrow.

Reply:
Recommended reading before making a decision wrt {full,first,last}_name: https://www.w3.org/International/questions/qa-personal-names.en.html

Reply:
I would not add a `full_name` field in database, this one can be formed in a frontend application when needed by using `first` and `last`. `username` is usually the name that the user will need to enter when authenticating, and should be therefore memorized by the user (using a hash in username is quite uncommon). And because one single real person might have several account with different permission, using a username with first-last-hash will not be good. Therefore I would let the user define his username (we can still let first-last as default) and we need to make sure that the username is unique. 
Due to technical user name, we should also add an optional field `description`, so that the user can have multiple account and can describe them if he want (as we have technical users in github, the `description` field would be a very nice feature).

Reply:
> [full_name] can be formed in a frontend application when needed by using first and last

Not in the general case, no. Might be good enough for most "Swiss" names though.

Reply:
To conclude the discussion: I go with `first_name` / `last_name` because
1. there might be a benefit in the future when connecting a user to a different user directory (eIAM?).
2. it is safe to assume that users are in Switzerland and have a sufficient workaround in case their name does not directly map to `first_name` / `last_name`.
3. A poll in the team showed that most are for this option.

I added the `username` as the primary key.